### PR TITLE
[stable/magic-namespace] Add TLS capabilities to Tiller deployment

### PR DESCRIPTION
**Pull request by acaire**
#### What this PR does / why we need it:
Adds TLS capabilities to the magic namespace installed Tiller, also:
* Fix Tiller container name
* Make additional service accounts optional

#### Special notes for your reviewer:
@lachie83 - Please review :)

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md

### CODE REVIEW

File: stable/magic-namespace/README.md [hunk 1]
```diff
@@ -140,6 +140,13 @@ reference the default `values.yaml` to understand further options.
 | `tiller.role.type` | Identify the name of the `Role` or `ClusterRole` that will be referenced in the role binding for Tiller's service account. There is seldom any reason to override this. | `admin` |
 | `tiller.includeService` | This deploys a service resource for Tiller. This is not generally needed. Please understand the security implications of this before overriding the default. | `false` |
 | `tiller.onlyListenOnLocalhost` | This prevents Tiller from binding to `0.0.0.0`. This is generally advisable to close known Tiller-based attack vectors. Please understand the security implications of this before overriding the default. | `true` |
+| `tls.enabled` | Whether to enable TLS encryption between Helm and Tiller. | `false` |
+| `tls.verify` | Whether to verify a remote certificate. | `true` |
+| `tls.certPath` | Path to mount certificates into the container. | `/etc/certs` |
+| `tls.secretName` | Provide TLS certs via an existing secret with Path to mount certificates into the container. Must include ca.crt, tls.crt and tls.key files| `nil` |
```
Comment (krancour) [inline stable/magic-namespace/README.md]: Is this option to use an existing secret mutually exclusive with the `tls.ca`, `tls.cert`, and `tols.key` options below?

If so, I feel like the documentation should make this clear.

File: stable/magic-namespace/templates/tiller-deployment.yaml [hunk 2]
```diff
@@ -30,7 +30,7 @@ spec:
     spec:
       serviceAccountName: tiller
       containers:
-        - name: {{ .Chart.Name }}
+        - name: tiller
```
Comment (krancour) [inline stable/magic-namespace/templates/tiller-deployment.yaml]: I'm not opposed to this, but it might be better as a separate PR since it's otherwise unrelated to the TLS capability.

**Comment by krancour** [comment]
Made a few comments, but this seems like a pretty solid improvement.

File: stable/magic-namespace/README.md (hunk 1 above)
Comment (krancour) [reply stable/magic-namespace/README.md]: Actually... now that I am thinking about it more, I'm not sure that the option to use an _existing_ secret makes much sense. The point of this chart is to _create_ and manage namespaces. This being the case, there cannot _be_ an existing secret in the namespace that you use this chart to manage.

**Review by krancour — CHANGES_REQUESTED** [review]
See comments.

File: stable/magic-namespace/README.md (hunk 1 above)
Comment (acaire) [reply stable/magic-namespace/README.md]: Hi @krancour , thanks for your review! It made me notice that I was specifying the options as `tls.foo` rather than `tiller.tls.foo`.  In relation to the custom secret, I too thought this when I assembled the PR, although the namespace creation is dependant on specifying _namespace_ i.e. `{{- if hasKey .Values "namespace" }}`, so it's feasible to use an existing namespace, with an existing secret.

File: stable/magic-namespace/templates/tiller-deployment.yaml (hunk 2 above)
Comment (acaire) [reply stable/magic-namespace/templates/tiller-deployment.yaml]: Thanks, I've moved this and the service account default changes to #12105 for consideration after this PR is merged.

**Comment by acaire** [comment]
> Made a few comments, but this seems like a pretty solid improvement.

Thanks @krancour - This is great.  Now we can use this chart to assign custom annotations for Kiam and set up isolated Tillers per namespace.  The only thing we're struggling to manage after these improvements is custom Roles i.e. we're passing this into our chart install:

```
roleBindings:
- name: admin
  role:
    kind: ClusterRole
    name: admin
  subject:
    kind: ServiceAccount
    name: tiller
```

But we fall short creating custom Roles that have additional privileges (such as Promethus Operator ServiceMonitors) i.e.:
```
Error: UPGRADE FAILED: Could not get information about the resource: servicemonitors.monitoring.coreos.com "my-custom-namespace" is forbidden: User "system:serviceaccount:my-custom-namespace:tiller" cannot get resource "servicemonitors" in API group "monitoring.coreos.com" in the namespace "my-custom-namespace"
```

This chart is the most logical place for us to create custom roles, do you think it's something you'd accept a PR for?

File: stable/magic-namespace/values.yaml [hunk 3]
```diff
@@ -23,6 +23,28 @@ tiller:
...
 
+  tls:
+    ## Enable TLS encryption between Helm and Tiller
+    enabled: false
+
+    ## Verify remote certificate
+    verify: true
+
+    ## The directory where Tiller's TLS certificates are located
+    certPath: /etc/certs
```
Comment (krancour) [inline stable/magic-namespace/values.yaml]: Any particular reason this is configurable? It seems to me that you could hardcode this as the value of the `TILLER_TLS_CERTS` environment variable and as the the `mountPath` for the volume and that would always work. I'm not sure what value the user could get out of overriding that.

File: stable/magic-namespace/README.md (hunk 1 above)
Comment (krancour) [reply stable/magic-namespace/README.md]: @acaire good point. I forgot I'd made that optional to support the edge case where the namespace already exists.

File: stable/magic-namespace/values.yaml (hunk 3 above)
Comment (acaire) [reply stable/magic-namespace/values.yaml]: I thought perhaps that some organisations may be enforcing a path to mount to i.e. `/ssl` for all containers but it's probably pretty far fetched so I'll make changes to hardcode them

Comment (acaire) [reply stable/magic-namespace/values.yaml]: @krancour ok that's hardcoded now.
